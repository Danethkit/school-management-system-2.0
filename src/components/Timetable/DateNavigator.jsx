import React from 'react'
import { Dialog,DialogActions,DialogContent,DialogTitle, Button, Toolbar, withStyles}
from '@material-ui/core'
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
import WeekPicker from '../Picker/WeekPicker'
const styles = theme => ({

    container1: {
        display: "flex",
        flexWrap: "wrap"
    },
    piker:{
        margin:15,
        // margin: theme.spacing(3)*2,
    },

    margin: {
        width:'100%',
        margin:0,
        textRendering: 'auto',
        letterSpacing: 'normal',
        textAlign:'center',
        fontSize:14,
        wordSpacing: 'normal',
        whiteSpace: 'pre-wrap',
        overflowWrap: 'break-word',

    },
    left: {
        flexGrow:1
    },
    right: {

    },
    middle: {
        flexGrow:1
    }

})
var moment = require("moment");
const weekOfYear = moment.utc().week();

const DateNavigator = ({classes, handleLastWeek, handleCurrentWeek, handleNextWeek, week, open,setOpen}) => {
    return (
        <Toolbar>
            <span className={classes.left}>
            <Button size="small" color="primary" className="button" variant="outlined" onClick={handleLastWeek} disabled={week === weekOfYear} >
                <KeyboardArrowLeft fontSize={'inherit'} />
                Last
            </Button>

            <Button size="small" color="primary" className="button" variant="outlined" onClick={handleCurrentWeek} disabled={week === weekOfYear} >
                Current
            </Button>

            <Button size="small"  color="primary" className="button" variant="outlined" onClick={handleNextWeek} >
                Next
                <KeyboardArrowRight fontSize={'inherit'}/>
            </Button>

            </span>
            <span className={classes.middle}>
                <b>Week{moment.utc().week(week).format('w') -17}( {moment.utc().week(week).weekday(0).format("MMM/DD")} - {moment.utc().week(week).weekday(6).format("MMM/DD")})</b>
            </span>
            <span className={classes.right}>


            <Button size="small" color="primary" className="button" variant="contained" onClick={()=>setOpen(true)} >
                    Duplicate
            </Button>
                <Dialog
                    disableBackdropClick
                    disableEscapeKeyDown
                    open={open}
                    aria-labelledby='Dialog-content'
                            >
                    <DialogTitle>Duplicate Timetable</DialogTitle>
                    <DialogContent id ='Dialog-content'>
                    <form className={classes.container1}>
                        <WeekPicker name ='Duplicate_from' className={classes.piker}/>
                        <WeekPicker name ='Duplicate_to' className={classes.piker}/>
                    </form>
                    </DialogContent>
                    <DialogActions>
                    <Button onClick={()=>setOpen(false)} color="secondary">
                        Cancel
                    </Button>
                    <Button onClick={()=>setOpen(false)} color="primary">
                        Ok
                    </Button>
                    </DialogActions>
                </Dialog>
            </span>
        </Toolbar>
    );
}
export default withStyles(styles)(DateNavigator)