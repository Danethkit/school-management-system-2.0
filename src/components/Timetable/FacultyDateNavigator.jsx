import React from 'react'
import {Button, Toolbar, withStyles}
from '@material-ui/core'
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
import moment from 'moment'

const styles = theme => ({

    container1: {
        display: "flex",
        flexWrap: "wrap"
    },

    piker:{
        margin:15,
    },

    margin: {
        width:'100%',
        margin:0,
        textRendering: 'auto',
        letterSpacing: 'normal',
        textAlign:'center',
        fontSize:14,
        wordSpacing: 'normal',
        whiteSpace: 'pre-wrap',
        overflowWrap: 'break-word',

    },

    left: {
        flexGrow:1
    },

    right: {

    },
    middle: {
        flexGrow:1
    }

})

const FacultyDateNavigator = ({classes, week, weekOfYear, handleLastWeek, handleCurrentWeek, handleNextWeek}) => {

    return (
        <Toolbar>
            <span className={classes.left}>
            <Button size="small" color="primary" className="button" variant="outlined" onClick={handleLastWeek} >
                <KeyboardArrowLeft fontSize={'inherit'} />
                Last
            </Button>

            <Button size="small" color="primary" className="button" variant="outlined" onClick={handleCurrentWeek} disabled={week === weekOfYear} >
                Current
            </Button>

            <Button size="small"  color="primary" className="button" variant="outlined" onClick={handleNextWeek} disabled={week === weekOfYear}>
                Next
                <KeyboardArrowRight fontSize={'inherit'}/>
            </Button>

            </span>
            <span className={classes.middle}>
                {
                    <b>{moment.utc().week(week).weekday(1).format("ddd MM/DD")+ '~' + moment.utc().week(week).weekday(6).format("ddd MM/DD")}</b>
                }
            </span>
        </Toolbar>
    );
}
export default withStyles(styles)(FacultyDateNavigator)