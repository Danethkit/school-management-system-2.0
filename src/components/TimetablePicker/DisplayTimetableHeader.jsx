import React, {useMemo} from'react'
import {Breadcrumbs,Typography} from '@material-ui/core';
function DisplayTimetableHeader({header, week})
{
  const memoValue = useMemo(()=>{
    return(
      <div style={{marginTop:30, padding:0}}>
      <Breadcrumbs separator="/" aria-label="Breadcrumb">
        <Typography color="textPrimary" variant="h5">{header.course}</Typography>
        <Typography color="textPrimary" variant="h5">{header.batch}</Typography>
        <Typography color="textPrimary" variant="h5">{header.semester}</Typography>
        <Typography color="textPrimary" variant="h5">{header.group}</Typography>
        <Typography color="textPrimary" variant="h5">{week}</Typography>
      </Breadcrumbs>
    </div>
  )
  }, [header, week])
  return memoValue
}
export default DisplayTimetableHeader
