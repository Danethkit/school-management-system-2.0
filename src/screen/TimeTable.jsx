import React, {useState, memo, useEffect} from "react";
import { Route, Switch } from "react-router-dom"
import AdminGenerateTimeTable from '../components/Timetable/AdminGenerateTimetable'
import FacultyTimeTable from '../components/Timetable/FacultyTimetableView'
import StudentTimeTableView from '../components/Timetable/StudentTimetableView'
import moment from 'moment'
import {connect} from 'react-redux'
import { onSemesterChange } from '../redux/ActionCreator/userBehavior'



const AttendanceScreen = (props) => {

  const {subjectInfo, course, batch, group, uid, dispatch} = props

  const [week, setWeek] = useState(moment.utc().week());
  const [weekStr, setWeekStr] = useState('');
  const [disableCurrentWeek, setDisableCurrentWeek] = useState(false)

  let currentWeek = ''
  let currentSem = ''
  if(Object.keys(subjectInfo).length !== 0 ){
    for(let semester in subjectInfo[course][batch]){
      for(let week of subjectInfo[course][batch][semester][group]['week']){
        let today = new Date()
        if(today >= new Date(week.startDate) && today <= new Date(week.endDate)){
          currentWeek = week
          currentSem = semester
          break
        }
      }
    }
  }
  // let endSemDate = useMemo(()=>{
  //   try{
  //     if(Object.keys(subjectInfo).length !== 0 && course){
  //       for(let semester in subjectInfo[course][batch]){
  //         for(let week of subjectInfo[course][batch][semester][group]['week']){
  //           let today = new Date()
  //           if(today >= new Date(week.startDate) && today <= new Date(week.endDate)){
  //             currentWeek = week.name
  //             setWeekStr(currentWeek)
  //             break
  //           }
  //         }
  //       }
  //       if(course && batch && semester){
  //         lastSemIndex = Object.keys(subjectInfo[course][batch])[Object.keys(subjectInfo[course][batch]).length -1]
  //         let lastWeekIndex = subjectInfo[course][batch][lastSemIndex][group]['week'].length -1
  //         try{
  //           endSemDate = subjectInfo[course][batch][semester][group]['week'][lastWeekIndex].endDate
  //         }catch{}
  //       }
  //     }
  //   } catch{ return ""}
  //   return endSemDate
  // }, [subjectInfo, batch, course, group])

  const handleChangeWeekStr = value => {
    if(!value) return
    // let weekIdnex = subjectInfo[course][batch][semester][group]['week'].findIndex(e => e.name === value)
    // ('check---------->', subjectInfo[course][batch][semester][group]['week'][weekIdnex].startDate);
    // setWeek(moment(subjectInfo[course][batch][semester][group]['week'][weekIdnex].startDate, 'YYYY-MM-DD').utc().week())
    setWeekStr(value);
  };

  const handleLastWeek = () => {
    let weekInt = parseInt(weekStr.split(' ')[1]) -1
    let weekstr = ''
    if(weekInt <10)
      weekstr = 'W 0'+weekInt
    else weekstr = 'W '+ weekInt
    setWeek(week - 1);
    setWeekStr(weekstr)
  };

  const handleCurrentWeek = () => {
    setWeek(moment(currentWeek.startDate, 'YYYY-MM-DD').week());
    setWeekStr(currentWeek.name)
    dispatch(onSemesterChange(currentSem))
  };

  useEffect(()=>{
    if(currentWeek.name) {
      setWeekStr(currentWeek.name)
      setWeek(moment(currentWeek.startDate, 'YYYY-MM-DD').week())
      dispatch(onSemesterChange(currentSem))
      setDisableCurrentWeek(false)
    }else {
      setDisableCurrentWeek(true)
    }
  }, [currentWeek])

  const handleNextWeek = () => {
    let weekInt = parseInt(weekStr.split(' ')[1]) +1
    let weekstr = ''
    if(weekInt <10)
      weekstr = 'W 0'+weekInt
    else weekstr = 'W '+ weekInt
    setWeekStr(weekstr)
    setWeek(week + 1);
  };

  const route = '/sms/timetable'
  return <>
  {
    !uid.uid ? null :
    uid.uid === 'admin' ?
    <Switch>
      <Route path = {`${route}/admin-create`}
        render={routerProps=>{
          return <AdminGenerateTimeTable
                  disableCurrentWeek ={disableCurrentWeek}
                  handleChangeWeekStr={handleChangeWeekStr}
                  handleLastWeek = {handleLastWeek}
                  handleCurrentWeek = {handleCurrentWeek}
                  handleNextWeek = {handleNextWeek}
                  week={week}
                  weekStr={weekStr}
                  {...props}
                  {...routerProps}/>
        }
      }
      />
      <Route path = {`${route}/faculty`}
       render={routerProps=>{
          return <FacultyTimeTable
            handleChangeWeekStr={handleChangeWeekStr}
            handleLastWeek = {handleLastWeek}
            handleCurrentWeek = {handleCurrentWeek}
            handleNextWeek = {handleNextWeek}
            week={week}
            weekStr={weekStr}
            {...props}
            {...routerProps}/>}}/>
      <Route path = {`${route}/student`} component={StudentTimeTableView} />
      <Route
        render={routerProps=>{
          return <AdminGenerateTimeTable
            disableCurrentWeek ={disableCurrentWeek}
            handleChangeWeekStr={handleChangeWeekStr}
            handleLastWeek = {handleLastWeek}
            handleCurrentWeek = {handleCurrentWeek}
            handleNextWeek = {handleNextWeek}
            week={week}
            weekStr={weekStr}
            {...props}
            {...routerProps}/>
        }
         } />
    </Switch>
    : uid.uid === 'faculty' ?
      <Route
       render={routerProps=>{
          return <FacultyTimeTable
            handleChangeWeekStr={handleChangeWeekStr}
            handleLastWeek = {handleLastWeek}
            handleCurrentWeek = {handleCurrentWeek}
            handleNextWeek = {handleNextWeek}
            week={week}
            weekStr={weekStr}
            {...props}
            {...routerProps}/>}}/> :
      <Route  component={StudentTimeTableView} />
  }
  </>
}
export default connect(state => ({
  subjectInfo: state.initData.subjectInfo,
  userIden: state.initData.userIden,
  uid: state.initData.uid,
  course: state.changePicker.course,
  batch: state.changePicker.batch,
  semester: state.changePicker.semester,
  group: state.changePicker.group
}))(memo(AttendanceScreen))
