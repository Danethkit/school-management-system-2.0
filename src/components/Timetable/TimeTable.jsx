import React, {useMemo, memo} from 'react'
import DisplayTimetableHeader from '../TimetablePicker/DisplayTimetableHeader'
import { Paper, Table, TableRow, TableCell, withStyles, Divider, Button, makeStyles, TableHead, TableBody, Box } from '@material-ui/core'
import moment from 'moment'
import InsertData from './InsertData'
const CustomTableCell = withStyles(theme => ({
    head: {
        backgroundColor: theme.palette.secondary.light,
        color: '#fff',
        fontSize: 14,
    },
    body: {
        fontSize: 14,
        paddingRight:5,
        paddingLeft:5,
        textAlign: 'center',
        margin:0,
    },
}))(TableCell);

const useStyles = makeStyles(theme => ({
    root: {
        width: '100%',
        marginTop: theme.spacing(3) *3 ,
        overflow: 'auto'
    },
    table: {
        minWidth: 540,
    },
    row: {
        '&:nth-of-type(odd)': {
            backgroundColor: '#ddd',
            fontSize: 14,
            paddingRight:5,
            paddingLeft:5,
            margin:0

        },
    },
    generateTimetable:{
        padding:0,
        marginTop:-60,
        marginBottom: theme.spacing(2)
    },
    container: {
        flexWrap: 'wrap',
        // margin:'-150 5 5 0',
    },
    submitButton: {
        display:"flex",
        justifyContent:"flex-end",
        padding: " 10px 0 15px 0",
    },
}))

const TimeTable = memo(({header, week, onDataInsert, sessions, facultyData}) => {
    const sortedSession = useMemo(()=> sessions.sort((a,b)=> {
        let timeA = [a.slice(0,5), ' ', a.slice(5,7)].join('')
        let timeB = [b.slice(0,5), ' ', b.slice(5,7)].join('')
        return new Date(`07/16/1999 ${timeA}`) - new Date(`07/16/1999 ${timeB}`)
    }), [sessions])

    const classes = useStyles()
    let columns = ['Session']
    for(let i=0; i<7; i++){
        columns.push(moment.utc().week(week).weekday(i).format("ddd MM/DD"))
    }
    return (
        <>
        <DisplayTimetableHeader header={header} week={week}/>
        <Box className={classes.generateTimetable} boxShadow={3}>
            <div className={classes.container}>
                <Paper className={classes.root}>
                    <Table className={classes.table}>
                        <TableHead>
                                <TableRow className={classes.row}>
                                    {
                                        columns.map(cell=><CustomTableCell align='center'  key={cell} >{cell}</CustomTableCell>)
                                    }
                                </TableRow>
                        </TableHead>
                        <TableBody>
                            {sortedSession.map(row => (
                                <TableRow className={classes.row} key={row} >
                                    {
                                        columns.map((cell,i)=> cell === 'Session' ? <CustomTableCell  align='center' component="th" scope="row" key={i}>{row}</CustomTableCell>
                                        : <CustomTableCell text-align= "center" key={i}><InsertData onChange={onDataInsert} row={row} col={cell} header={header} facultyData={facultyData}/></CustomTableCell>)
                                    }
                                </TableRow>
                            ))}
                        </TableBody>
                    </Table>
                </Paper>
            </div>

            <Divider style={{marginLeft:15,marginRight:15}}/>
            <div className={classes.submitButton} >
                <Button size="medium" variant="outlined" color="primary" disabled={true} style={{marginLeft:15,marginRight:15}} >Edit</Button>
                <Button size="medium" variant="outlined" color="primary"  >Discard</Button>
                <Button size="medium"  color="primary" style={{marginLeft:15,marginRight:15}}  variant="contained">Save</Button>
            </div>
        </Box>
        </>
    )
})
export default TimeTable