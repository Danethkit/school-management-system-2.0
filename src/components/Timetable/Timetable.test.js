import React from 'react'
import {shallow} from 'enzyme'
import DateNavigator from './DateNavigator'
import DisplayTimetableHeader from './DisplayTimetableHeader'
import FacultyDateNavigator from './FacultyDateNavigator'
import TimeTable from './TimeTable'
describe('Timetable', () => {
    describe('DateNavigator', () => {
        let Date_Navigator
        beforeEach(()=>{
            Date_Navigator = shallow(<DateNavigator/>)
        })
        it('should render without errors', () => {
            expect(Date_Navigator).toMatchSnapshot()
        });
    });
    describe('DisplayTimetableHeader', () => {
        let DisplayTime_tableHeader
        beforeEach(()=>{
            DisplayTime_tableHeader = shallow(<DisplayTimetableHeader/>)
        })
        it('should render without errors', () => {
            expect(DisplayTime_tableHeader).toMatchSnapshot()
        });
    });
    describe('FacultyDateNavigator', () => {
        let FacultyDate_Navigator
        beforeEach(()=>{
            FacultyDate_Navigator = shallow(<FacultyDateNavigator/>)
        })
        it('should render without errors', () => {
            expect(FacultyDate_Navigator).toMatchSnapshot()
        });
    });
    describe('TimeTable', () => {
        let Time_Table
        beforeEach(()=>{
            Time_Table = shallow(<TimeTable/>)
        })
        it('should render without errors', () => {
            expect(Time_Table).toMatchSnapshot()
        });
    });
});
