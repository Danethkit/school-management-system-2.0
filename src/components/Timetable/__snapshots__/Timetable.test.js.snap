// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timetable DateNavigator should render without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(WithStyles) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "container1": "DateNavigator-container1-1",
        "left": "DateNavigator-left-4",
        "margin": "DateNavigator-margin-3",
        "middle": "DateNavigator-middle-6",
        "piker": "DateNavigator-piker-2",
        "right": "DateNavigator-right-5",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "container1": "DateNavigator-container1-1",
          "left": "DateNavigator-left-4",
          "margin": "DateNavigator-margin-3",
          "middle": "DateNavigator-middle-6",
          "piker": "DateNavigator-piker-2",
          "right": "DateNavigator-right-5",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Timetable DisplayTimetableHeader should render without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayTimetableHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": null,
    },
    "ref": null,
    "rendered": null,
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": null,
      },
      "ref": null,
      "rendered": null,
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Timetable FacultyDateNavigator should render without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(WithStyles) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "container1": "FacultyDateNavigator-container1-7",
        "left": "FacultyDateNavigator-left-10",
        "margin": "FacultyDateNavigator-margin-9",
        "middle": "FacultyDateNavigator-middle-12",
        "piker": "FacultyDateNavigator-piker-8",
        "right": "FacultyDateNavigator-right-11",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "container1": "FacultyDateNavigator-container1-7",
          "left": "FacultyDateNavigator-left-10",
          "margin": "FacultyDateNavigator-margin-9",
          "middle": "FacultyDateNavigator-middle-12",
          "piker": "FacultyDateNavigator-piker-8",
          "right": "FacultyDateNavigator-right-11",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Timetable TimeTable should render without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Component) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
